File: apps/api/app/compliance_states.py
"""Shared lifecycle classification. Closure is distinct from successful completion."""
CLOSED_STATES = frozenset({"COMPLETED", "CANCELLED", "NOT_APPLICABLE"})
EXCLUDED_STATES = CLOSED_STATES - {"COMPLETED"}


def is_open(status: str) -> bool:
    return status not in CLOSED_STATES


def counts_toward_completion(status: str) -> bool:
    return status not in EXCLUDED_STATES

File: apps/web/lib/compliance-states.ts
// Keep consistent with the server lifecycle classification.
export const closedComplianceStates = new Set(["COMPLETED", "CANCELLED", "NOT_APPLICABLE"]);
export const isOpenCompliance = (status: string) => !closedComplianceStates.has(status);
export const countsTowardCompletion = (status: string) => !["CANCELLED", "NOT_APPLICABLE"].includes(status);
